NachoSpigot-API/src/main/java/com/google/common/util/concurrent/Futures.java

```diff
@@ -1450,5 +1450,221 @@ private static void wrapAndThrowUnchecked(Throwable cause) {
+   * ListenableFuture<Integer> faultTolerantFuture = Futures.catchingAsync(
+   *     fetchCounterFuture, FetchException.class, x -> immediateFuture(0));
+   * }</pre>
+   *
+   * <p>The fallback can also choose to propagate the original exception when desired:
+   *
+   * <pre>{@code
+   * ListenableFuture<Integer> fetchCounterFuture = ...;
+   *
+   * // Falling back to a zero counter only in case the exception was a
+   * // TimeoutException.
+   * ListenableFuture<Integer> faultTolerantFuture = Futures.catchingAsync(
+   *     fetchCounterFuture,
+   *     FetchException.class,
+   *     e -> {
+   *       if (omitDataOnFetchFailure) {
+   *         return immediateFuture(0);
+   *       }
+   *       throw e;
+   *     });
+   * }</pre>
+   *
+   * <p>This overload, which does not accept an executor, uses {@code directExecutor}, a dangerous
+   * choice in some cases. See the discussion in the {@link ListenableFuture#addListener
+   * ListenableFuture.addListener} documentation. All its warnings about heavyweight listeners are
+   * also applicable to heavyweight functions passed to this method. (Specifically, {@code
+   * directExecutor} functions should avoid heavyweight operations inside {@code
+   * AsyncFunction.apply}. Any heavyweight operations should occur in other threads responsible for
+   * completing the returned {@code Future}.)
+   *
+   * @param input the primary input {@code Future}
+   * @param exceptionType the exception type that triggers use of {@code fallback}. The exception
+   *     type is matched against the input's exception. "The input's exception" means the cause of
+   *     the {@link ExecutionException} thrown by {@code input.get()} or, if {@code get()} throws a
+   *     different kind of exception, that exception itself. To avoid hiding bugs and other
+   *     unrecoverable errors, callers should prefer more specific types, avoiding {@code
+   *     Throwable.class} in particular.
+   * @param fallback the {@link AsyncFunction} to be called if {@code input} fails with the expected
+   *     exception type. The function's argument is the input's exception. "The input's exception"
+   *     means the cause of the {@link ExecutionException} thrown by {@code input.get()} or, if
+   *     {@code get()} throws a different kind of exception, that exception itself.
+   * @since 19.0 (similar functionality in 14.0 as {@code withFallback})
+   * @deprecated Use {@linkplain #catchingAsync(ListenableFuture, Class, AsyncFunction, Executor)
+   *     the overload that requires an executor}. For identical behavior, pass {@link
+   *     MoreExecutors#directExecutor}, but consider whether another executor would be safer, as
+   *     discussed in the {@link ListenableFuture#addListener ListenableFuture.addListener}
+   *     documentation. This method is scheduled to be removed in July 2018.
+   */
+  @CanIgnoreReturnValue
+  @Deprecated
+  @com.google.errorprone.annontations.DoNotCall
```

Comment:
`annontations`. Doesn't seem right?

Reply:
Thanks haha
```suggestion
  @com.google.errorprone.annotations.DoNotCall
```

Reply:
Oh that's actually what caused the error (I didn't look at the error message in the workflow). Could you fix it?